specs/interop/derivation.md

```diff
@@ -87,17 +91,185 @@ The interop network upgrade timestamp defines the timestamp at which all functio
+# 0x88c6b48354c367125a59792a93a7b60ad7cd66e516157dbba16558c68a46d3cb
+```
+
+### L2ToL2CrossDomainMessenger Deployment
+<!-- Generated with: ./scripts/run_gen_predeploy_docs.sh --optimism-repo-path ../optimism --fork-name Interop --contract-name L2ToL2CrossDomainMessenger --from-address 0x4220000000000000000000000000000000000001 --from-address-nonce 0 --git-commit-hash 71c460ec7c7c05791ddd841b97bcb664a1f0c753 --eth-rpc-url https://optimism.rpc.subquery.network/public --proxy-address 0x4200000000000000000000000000000000000023 --copy-contract-bytecode true -->
+
+The `L2ToL2CrossDomainMessenger` contract is deployed.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4220000000000000000000000000000000000001`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `nonce`: `0`
+- `gasLimit`: `1100000`
+- `data`: `0x0x6080604052348015600e575f80fd5b...` ([full bytecode](../static/bytecode/interop-l2-to-l2-cross-domain-messenger-deployment.txt))
+- `sourceHash`: `0xf5484697c7a9a791db32a3bf0763bf2ba686c77ae7d4c0a5ee8c222a92a8dcc2`,  
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: L2ToL2CrossDomainMessenger Deployment"`
+
+This results in the Interop L2ToL2CrossDomainMessenger contract being deployed to
+`0x0D0eDd0ebd0e94d218670a8De867Eb5C4d37cadD`, to verify:
+
+```bash
+cast compute-address --nonce=0 0x4220000000000000000000000000000000000001
+Computed Address: 0x0D0eDd0ebd0e94d218670a8De867Eb5C4d37cadD
+```
+
+Verify `sourceHash`:
+
+```bash
+cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: L2ToL2CrossDomainMessenger Deployment"))
+# 0xf5484697c7a9a791db32a3bf0763bf2ba686c77ae7d4c0a5ee8c222a92a8dcc2
+```
+
+Verify `data`:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+jq -r ".bytecode.object" packages/contracts-bedrock/forge-artifacts/L2ToL2CrossDomainMessenger.sol/L2ToL2CrossDomainMessenger.json
+```
+
+This transaction MUST deploy a contract with the following code hash  
+`0x458925c90ec70736600bef3d6529643a0e7a0a848e62626d61314c057b4a71a9`.
+
+To verify the code hash:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+cast k $(jq -r ".deployedBytecode.object" packages/contracts-bedrock/forge-artifacts/L2ToL2CrossDomainMessenger.sol/L2ToL2CrossDomainMessenger.json)
+```
+
+### L2ToL2CrossDomainMessenger Proxy Update
+
+This transaction updates the L2ToL2CrossDomainMessenger Proxy ERC-1967
+implementation slot to point to the new L2ToL2CrossDomainMessenger deployment.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000023` (L2ToL2CrossDomainMessenger Proxy)
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
```

Comment:
I would like to know how do we verify that the `gasLimit` is correct each time? 
I suppose we have tests that perform the upgrade and then we copy the `gasLimit`? If yes, where are the tests to make sure the `gasLimit` is copied correctly for each: 
```
- `gasLimit`: `50,000`
- `gasLimit`: `1100000`
- `gasLimit`: `50,000`
- `gasLimit`: `420000`
```

Reply:
There are tests in the source code that assert the post state of the txs and would catch the revert